src/App.tsx

```diff
@@ -15,71 +21,100 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+    title: '',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  addItem = (item: string) => {
+    this.setState({
+      selectedGood: item,
+      title: '',
+    });
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  removeItem = () => {
+    this.setState({
+      selectedGood: '',
+      title: 'No goods selected',
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  handleItem = (item: string) => {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    if (selectedGood === item) {
+      this.removeItem();
+    } else {
+      this.addItem(item);
+    }
+  }
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood, title } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    function isItemSelected(item: string) {
+      return selectedGood === item;
+    }
 
-        <tr data-cy="Good">
-          <td>
+    return (
+      <main className="section container">
+        {title ? (
+          <h1 className="title">{title}</h1>
+        ) : (
```

Comment:
you can check here selectedGood variable, and print needed text if there is no selected good
title variable is redundant

Reply:
done